components/notebook-controller/Dockerfile

```diff
@@ -30,14 +32,12 @@ WORKDIR /workspace/notebook-controller
 ## Build the kf-notebook-controller
 USER root
 
-RUN if [ -z ${CACHITO_ENV_FILE} ]; then \
-       go mod download all; \
-    else \
-       source ${CACHITO_ENV_FILE}; \
-    fi
-
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -mod=mod \
-        -o ./bin/manager main.go
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN if [ -z ${CACHITO_ENV_FILE} ]; then go mod download; else source ${CACHITO_ENV_FILE}; fi && \
+  CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -tags strictfipsruntime -a -o ./bin/manager main.go
```

Comment:
> Why TARGETARCH doesn't have a default value similarly as TARGETOS?

That's explained in the comment above. No default means default to the arch of the container.

Reply:
thanks, I didn't pay enough attention to the comment